### ISSUE TITLE
Output for keys not present in Hash constructed using to_h is size of Hash

### ISSUE THREAD

**Issue by Confusion** [oracle/truffleruby#1698]
TruffleRuby 19.0.0, JDK 11.0.3

The following code shows that the output for keys not present in the Hash is the size of the Hash.
```
$ truffleruby -e "h = [[1, 1], ['a', 'a']].to_h; puts h.class; puts h['whatever']"
Hash
2
```

**Comment by chrisseaton** [comment]
Yes, what a strange bug!

It looks like someone mistook the initialiser of `Hash` as setting a capacity, when it's actually the default value.

https://github.com/oracle/truffleruby/blob/59a79086eec7ddbcab3b1f100c8e41f4b18c8d70/src/main/ruby/truffleruby/core/array.rb#L1185

And the behaviour you're testing isn't covered by the Ruby specs which is why we haven't noticed before.

https://github.com/oracle/truffleruby/blob/59a79086eec7ddbcab3b1f100c8e41f4b18c8d70/spec/ruby/core/array/to_h_spec.rb

Thanks, this should be a very easy fix. We'll get it in with new specs to describe the behaviour immediately.

**Comment by Confusion** [comment]
Seems like a good case for a roundtrip test: `hash.to_a.to_h` should equal the original `hash` in all respects, including the default.

**Comment by eregon** [comment]
Actually `hash.to_a.to_h` does not preserve the default on MRI (the Array doesn't keep that information):
```
$ ruby -ve 'h=Hash.new(42); p({a: 1}.to_a.to_h.default)'
ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-linux]
nil
```
But indeed we should test the `default` and `default_proc` of `Array#to_h` is always `nil`.

**Comment by chrisseaton** [comment]
This is fixed in 2f8100c1f62caacc0e4f616ff9d05e98649c895b.
